index.bs

```diff
@@ -52,10 +66,169 @@ dictionary StorageBucketOptions {
 </xmp>
 
 <h3 id="storage-bucket-open">Creating a bucket</h3>
+
+<div algorithm>
+
+The <dfn method for="StorageBucketManager">open(|name|, |options|)</dfn> method steps are:
+
+1. Let |environment| be [=/this=]'s [=/relevant settings object=].
+
+1. Let |storageKey| be the result of running [=obtain a storage key=] given |environment|.
+
+1. If |storageKey| is failure, then [=exception/throw=] a "{{SecurityError}}" {{DOMException}} and abort these steps.
+
+1. Let |p| be [=a new promise=].
+
+1. Run the following steps [=in parallel=]:
+
+    1. If the result of [=validate a bucket name=] with |name| is failure, then [=/reject=] |p| with an {{InvalidCharacterError}} and abort these steps.
+
+    1. Let |r| be the result of running [=open a bucket=] with |storageKey|, |name|, and |options|.
+
+    1. If |r| is failure, then [=reject=] |p| with a {{TypeError}} and abort these steps.
+
+    1. [=/Resolve=] |p| with |r|.
+
+1. Return |p|.
+
+</div>
+
+<div algorithm>
+
+To <dfn>open a bucket</dfn> for a |storageKey| given a bucket |name| and optional |options|, run the following steps:
+
+1. Let |expires| be |options|["{{StorageBucketOptions/expires}}"] if it exists, otherwise undefined.
+
+1. If |expires| is not undefined, and is less than or equal to now, then return failure.
+
+1. Let |quota| be |options|["{{StorageBucketOptions/quota}}"] if it exists, otherwise undefined.
+
+1. If |quota| is less than or equal to zero, then return failure.
+
+1. Let |persisted| be false.
+
+1. If |options|["{{StorageBucketOptions/persisted}}"] is true, then:
+
+   1. Let |permission| be the result of [=/requesting permission to use=] "<code>persistent-storage</code>".
+
+   1. If |permission| is "{{PermissionState/granted}}", then set |persisted| to true.
+
+1. Let |bucket| be the [=/storage bucket=] named |name| in |storageKey| or null otherwise.
+
+1. If |bucket| is non-null and |bucket|'s [=bucket expiration=] is less than or equal to now, then:
+
+   1. Remove |bucket|.
+
+   1. Set |bucket| to null.
+
+1. If |bucket| is null, then:
+
+   1. Let |bucket| be a new [=/storage bucket=] with name |name|
+
+   1. Set |bucket|'s [=bucket durability|durability=] to |options|["{{StorageBucketOptions/durability}}"] if it exists.
+
+   1. Set |bucket|'s [=bucket quota|quota=] to |quota|.
+
+1. If |persisted| is true, set |bucket|'s [=/bucket mode=] to "<code>persistent</code>".
+
+1. Set |bucket|'s [=bucket expiration|expiration=] to |expires|.
+
+1. Let |storageBucket| be a new {{StorageBucket}}.
+
+1. Set |storageBucket|'s [=/storage bucket=] to |bucket|.
+
+1. Return |storageBucket|.
+
+</div>
+
+<div algorithm>
+
+To <dfn>validate a bucket name</dfn> given string |name|, run the following steps:
+
+1. If |name| contains any character that is not [=ASCII lower alpha=], [=ASCII digit=], U+005F (_), or U+002D(-), then return failure.
```

Comment:
I think this should probably say `[=code point=]` instead of character.  For example, `[=ASCII lower alpha=]` is defined in terms of code points.

(Also, if we were going to use it I think the technical term to use instead of character is `[=code unit=]`.)